integration/bosh_links_test.go

```diff
@@ -68,4 +67,37 @@ var _ = Describe("BOSHLinks", func() {
 			})
 		})
 	})
+
+	Context("when deployment has explicit external link dependencies", func() {
+		BeforeEach(func() {
+			natsConfigMap := env.Catalog.NatsConfigMap(deploymentName)
+			tearDown, err := env.CreateConfigMap(env.Namespace, natsConfigMap)
+			Expect(err).NotTo(HaveOccurred())
+			tearDowns = append(tearDowns, tearDown)
+
+			natsSecret := env.Catalog.NatsSecret(deploymentName)
+			tearDown, err = env.CreateSecret(env.Namespace, natsSecret)
+			Expect(err).NotTo(HaveOccurred())
+			tearDowns = append(tearDowns, tearDown)
+
+			natsPod := env.Catalog.NatsPod(deploymentName)
+			tearDown, err = env.CreatePod(env.Namespace, natsPod)
+			Expect(err).NotTo(HaveOccurred())
+			tearDowns = append(tearDowns, tearDown)
+
+			natsService := env.Catalog.NatsService(deploymentName)
+			tearDown, err = env.CreateService(env.Namespace, natsService)
+			Expect(err).NotTo(HaveOccurred())
+			tearDowns = append(tearDowns, tearDown)
+
+			boshManifest = env.BOSHManifestSecret(manifestRef, bm.NatsSmokeTestWithExternalLinks)
+		})
+
+		It("creates a secret for each link", func() {
```

Comment:
The test checks for pods however? Should it check for secrets? Or should this read `"waits for pods from the instance gr out"`?